### PR TITLE
Feature/31 edit covid test

### PR DESCRIPTION
**Pull request by ankestein**
Add backend and frontend functionality to edit a Covid Test

### CODE REVIEW

File: backend/src/main/java/de/neuefische/backend/service/CovidTestService.java [hunk 1]
```diff
@@ -44,5 +44,14 @@ public CovidTest addCovidTest(CovidTestDto covidTestDto) {
             return covidTestRepo.save(covidTest);
         }
     }
-    
+
```
Comment (belluma) [inline backend/src/main/java/de/neuefische/backend/service/CovidTestService.java:47]:
```suggestion
import static CovidTestMapper.*
```
makes the code further down more readable

File: backend/src/test/java/de/neuefische/backend/controller/CovidTestControllerTest.java [hunk 2]
```diff
@@ -107,6 +109,79 @@ void getCovidTestsTest() {
...
+        // GIVEN
+        CovidTestDto covidTestDto = CovidTestDto.builder()
+                .id("unknownID222")
+                .testType(TestType.ANTIGEN_NASAL)
+                .dateTime(LocalDateTime.of(2021, 11, 15, 7, 5))
+                .result(Result.NEGATIVE)
+                .build();
+
+        // WHEN
+        ResponseEntity<CovidTest> response = testRestTemplate.exchange("/api/covid-test/unknownID222", HttpMethod.PUT, new HttpEntity<>(covidTestDto, headers), CovidTest.class);
```
Comment (belluma) [inline backend/src/test/java/de/neuefische/backend/controller/CovidTestControllerTest.java:156]:
```suggestion
        ResponseEntity<ApiError> response = testRestTemplate.exchange("/api/covid-test/unknownID222", HttpMethod.PUT, new HttpEntity<>(covidTestDto, headers), ApiError.class);
```

File: backend/src/test/java/de/neuefische/backend/controller/CovidTestControllerTest.java [hunk 3]
```diff
@@ -107,6 +109,79 @@ void getCovidTestsTest() {
...
+
+        // THEN
+        assertThat(response.getStatusCode(), is(HttpStatus.NOT_FOUND));
+    }
```
Comment (belluma) [inline backend/src/test/java/de/neuefische/backend/controller/CovidTestControllerTest.java:160]:
```suggestion
   assertThat(response.getBody().getExceptionMessage(), is("the message of your exception")
     }
```

File: frontend/src/components/CovidTestCard.js [hunk 4]
```diff
@@ -12,45 +39,83 @@ export default function CovidTestCard({covidTest}) {
...
+
+	const handleClickClose = () => {
+		setOpen(false);
+	};
+
+	const handleSave = () => {
+		editCovidTest(covidTestToEdit, token).catch(console.error);
+		setNewCovidTest(initialCovidTest);
+		getCovidTests(token)
+			.then((covidTests) => setAllCovidTests(covidTests))
```
Comment (belluma) [inline frontend/src/components/CovidTestCard.js:55]:
```suggestion
			.then(setAllCovidTests)
```

File: frontend/src/components/CovidTestCard.js [hunk 5]
```diff
@@ -12,45 +39,83 @@ export default function CovidTestCard({covidTest}) {
...
+						variant='overline'
+						style={{color: '#303030', fontSize: 10}}
+					>
+						{testTypes[covidTest.testType]}
+					</Typography>
+					<Typography
+						variant='subtitle2'
+						style={{color: '#233d4dff', marginBottom: 8, fontSize: 10}}
+					>
+						{moment(covidTest.dateTime).format('llll')}
```
Comment (belluma) [inline frontend/src/components/CovidTestCard.js:74]: consider this
https://momentjs.com/docs/#/-project-status/

File: frontend/src/pages/CovidTests.js [hunk 6]
```diff
@@ -3,7 +3,13 @@ import {useHistory} from 'react-router-dom';
 import CovidTestCard from '../components/CovidTestCard';
 import AddButton from '../components/AddButton';
 
-export default function CovidTests({allCovidTests}) {
+export default function CovidTests({
```
Comment (belluma) [inline frontend/src/pages/CovidTests.js:6]:
```suggestion
export default function CovidTests({allCovidTests, cardProps})
```

File: frontend/src/pages/CovidTests.js [hunk 7]
```diff
@@ -18,7 +24,15 @@ export default function CovidTests({allCovidTests}) {
 
 			<CardContainer>
 				{allCovidTests.map((covidTest) => {
-					return <CovidTestCard covidTest={covidTest} />;
+					return (
+						<CovidTestCard
+							covidTest={covidTest}
+							setAllCovidTests={setAllCovidTests}
```
Comment (belluma) [inline frontend/src/pages/CovidTests.js:30]:
```suggestion
							{...cardProps}
```

File: frontend/src/pages/CovidTests.js [hunk 8]
```diff
@@ -18,7 +24,15 @@ export default function CovidTests({allCovidTests}) {
...
+							newCovidTest={newCovidTest}
```
Comment (belluma) [inline frontend/src/pages/CovidTests.js:31]:
```suggestion
							
```

File: frontend/src/pages/CovidTests.js [hunk 9]
```diff
@@ -18,7 +24,15 @@ export default function CovidTests({allCovidTests}) {
...
+							setNewCovidTest={setNewCovidTest}
```
Comment (belluma) [inline frontend/src/pages/CovidTests.js:32]:
```suggestion
							
```

File: frontend/src/pages/CovidTests.js [hunk 10]
```diff
@@ -18,7 +24,15 @@ export default function CovidTests({allCovidTests}) {
...
+							initialCovidTest={initialCovidTest}
```
Comment (belluma) [inline frontend/src/pages/CovidTests.js:33]:
```suggestion
							
```

File: frontend/src/pages/NewCovidTest.js [hunk 11]
```diff
@@ -15,16 +15,24 @@ import {
...
-	const [newCovidTest, setNewCovidTest] = useState(initialCovidTest);
+export default function NewCovidTest({
+	setAllCovidTests,
+	newCovidTest,
+	setNewCovidTest,
+	initialCovidTest,
+}) {
 	const [showError, setShowError] = useState(false);
+	const {token} = useContext(AuthContext);
 	const history = useHistory();
```
Comment (belluma) [inline frontend/src/pages/NewCovidTest.js:37]: rendering your fromcontrollabels dynamically would save you many lines of code and make it much easier to add/remove elements.
sth like this: 
const values = ['ANTIGEN_LOLLIPOP', ...]
const labels = ['Antigen test - lollipop', ...]
<RadioGroup>{values.map((v, i) => <FormControlLabel value=v control={<Radio/>} label=labels[i] />)}

**Review by belluma — APPROVED** [review]

File: frontend/src/components/CovidTestCard.js (hunk 5 above)
Comment (ankestein) [reply frontend/src/components/CovidTestCard.js:74]: Interesting!

**Pull request merged**